native/cocos/platform/android/adpf_manager.cpp

```diff
@@ -249,24 +279,85 @@ void ADPFManager::SetThermalListener(thermalStateChangeListener listener) {
 // Indicates the start and end of the performance intensive task.
 // The methods call performance hint API to tell the performance
 // hint to the system.
-void ADPFManager::BeginPerfHintSession() { perfhintsession_start_ = std::chrono::high_resolution_clock::now(); }
+void ADPFManager::BeginPerfHintSession() {
+    perf_start_ = std::chrono::steady_clock::now();
+}
 
 void ADPFManager::EndPerfHintSession(jlong target_duration_ns) {
-    auto current_clock = std::chrono::high_resolution_clock::now();
-    auto duration = current_clock - perfhintsession_start_;
-    frame_time_ns_ = std::chrono::duration_cast<std::chrono::nanoseconds>(duration).count();
+#if __ANDROID_API__ >= 33
+    auto perf_end = std::chrono::steady_clock::now();
+    auto dur = std::chrono::duration_cast<std::chrono::nanoseconds>(perf_end - perf_start_).count();
+    int64_t actual_duration_ns = static_cast<int64_t>(dur);
+    APerformanceHint_reportActualWorkDuration(hint_session_, actual_duration_ns);
+    APerformanceHint_updateTargetWorkDuration(hint_session_, target_duration_ns);
+#else
+    auto perf_end = std::chrono::steady_clock::now();
+    auto dur = std::chrono::duration_cast<std::chrono::nanoseconds>(perf_end - perf_start_).count();
+    jlong actual_duration_ns = static_cast<jlong>(dur);
     if (obj_perfhint_session_) {
-        jlong duration_ns = std::chrono::duration_cast<std::chrono::nanoseconds>(
-                                duration * 100000000)
-                                .count();
         auto *env = cc::JniHelper::getEnv();
 
         // Report and update the target work duration using JNI calls.
         env->CallVoidMethod(obj_perfhint_session_, report_actual_work_duration_,
-                            duration_ns);
+                            actual_duration_ns);
         env->CallVoidMethod(obj_perfhint_session_, update_target_work_duration_,
                             target_duration_ns);
     }
+#endif
+}
+void ADPFManager::AddThreadIdToHintSession(int32_t tid) {
+    thread_ids_.push_back(tid);
+    auto data = thread_ids_.data();
+
+    RegisterThreadIdsToHintSession();
+}
+
+void ADPFManager::RemoveThreadIdFromHintSession(int32_t tid) {
+    thread_ids_.erase(std::remove(thread_ids_.begin(), thread_ids_.end(), tid), thread_ids_.end());
```

Comment:
The usage of std::move() depends on c++ 20?

Reply:
Should be available since the ANSI C++, it is the one from <algorithm> https://[cplusplus.com/reference/algorithm/remove/](https://cplusplus.com/reference/algorithm/remove/)

Reply:
It is wired. What i referred is the doc: https://en.cppreference.com/w/cpp/algorithm/remove. It seems c++17 only supports
```c++
template< class ExecutionPolicy, class ForwardIt, class T >
ForwardIt remove( ExecutionPolicy&& policy,
                  ForwardIt first, ForwardIt last, const T& value );
```

Reply:
Sorry, i missed the `until c++20` for 
```c++
template< class ForwardIt, class T >
ForwardIt remove( ForwardIt first, ForwardIt last, const T& value );
```